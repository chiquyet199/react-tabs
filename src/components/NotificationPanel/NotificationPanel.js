import styled from "styled-components/native";
import {isWeb} from "../../utils/common";

export const Container = styled.View`
  position: ${({topbar}) =>
    topbar === "true" && isWeb ? "absolute" : "relative"};
  left: 0;
  top: ${({topbar}) => (topbar === "true" && isWeb ? "100%" : "0")};
  background-color: ${({color, theme}) => color || theme.colors.none};
  opacity: 0.8;
  min-height: 38px;
  width: 100%;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  padding-horizontal: 15px;
  margin-top: ${({topmargin}) =>
    topmargin === 0 || topmargin ? `${topmargin}px` : "-15px"};
  margin-bottom: ${({bottommargin}) =>
    bottommargin === 0 || bottommargin ? `${bottommargin}px` : "0"};
`;

export const NotificationText = styled.Text`
  padding: 0;
  color: white;
  flex-grow: 3;
  color: ${({color, theme}) => color || theme.colors.lightSteel};
`;

export const TextWrapper = styled.View`
  flex-grow: 2;
  padding-left: 15px;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  flex-wrap: nowrap;
  padding: 10px;
`;
