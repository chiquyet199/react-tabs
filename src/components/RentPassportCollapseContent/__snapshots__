/Users/edward/Desktop/canopy-frontend/packages/canopy-frontend-common/src/components/RentPassportCollapseContent/__native_snapshots__