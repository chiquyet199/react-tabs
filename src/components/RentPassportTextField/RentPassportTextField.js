import React from "react";
import styled from "styled-components/native";
import {colors} from "../../constants/theme";
import {bool} from "prop-types";

export const FlexContainer = styled.View`
  flex: auto;
  flex-direction: row;
  flex-wrap: wrap;
  width: ${props => props.width}%;
  padding-vertical: 10px;
`;

export const PageHeader = styled.Text`
  font-size: ${({fontSize}) => fontSize || "18"}px;
  display: flex;
  font-style: ${({fontStyle}) => fontStyle || "normal"};
  width: ${({align}) => (align === "right" ? "auto" : "100%")};
  color: ${({error}) =>
    error === "true" ? colors.alertRed : colors.canopySteel};
  align-self: flex-end;
  align-items: flex-end;
  justify-content: ${({align}) =>
    align === "right" ? "space-between" : "flex-start"};
`;

export const TextFieldHeading = styled.Text`
  font-size: 14px;
  font-weight: bold;
  color: ${({error}) =>
    error === "true" ? colors.alertRed : colors.canopySteel};
  width: 100%;
  display: flex;
`;

class RentPassportTextField extends React.Component {
  joinN = value => {
    if (Array.isArray(value) && value) {
      return value.filter(n => n).join("\n");
    } else if (typeof value === "string" && value.trim().length) {
      return value;
    } else if (typeof value === "number" || typeof value === "object") {
      return value;
    }

    return "";
  };

  render() {
    const {title, value, width, fontStyle, fontSize, align, error} = this.props;
    const errorProp = error ? error.toString() : undefined;

    return (
      <FlexContainer width={width} align={align}>
        {title && (
          <TextFieldHeading error={errorProp} align={align}>
            {title}
          </TextFieldHeading>
        )}
        <PageHeader
          fontStyle={fontStyle}
          fontSize={fontSize}
          align={align}
          error={errorProp}
          textAlignVertical="bottom"
        >
          {this.joinN(value)}
        </PageHeader>
      </FlexContainer>
    );
  }
}

RentPassportTextField.propTypes = {
  error: bool,
};

export default RentPassportTextField;
