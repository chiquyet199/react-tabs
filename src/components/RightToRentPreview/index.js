import React, {Component} from "react";
import {themed} from "../../utils/common";
import {connect} from "react-redux";
import {
  Container,
  StyledText,
  ScanningMetadata,
} from "./RightToRentPreview.styles";
import {arrayOf, func, oneOf, shape, string} from "prop-types";
import moment from "moment";
import {fetchUserById} from "../../actions/users";

export class RightToRentPreview extends Component {
  componentDidMount() {
    const {
      getUser,
      passport: {
        data: {seenBy},
      },
    } = this.props;
    getUser(seenBy);
  }

  getScanningMetadata() {
    const {
      passport: {
        data: {seenAt},
        translations: {seenBy},
      },
      scannedByName,
    } = this.props;

    return `${seenBy}${scannedByName}, ${moment(seenAt).format("DD.MM.YYYY")}`;
  }

  render() {
    const {passport, theme, globaltheme, scannedByName} = this.props;
    const hasRightToRent = passport.data.documents.length > 0;
    const {
      translations,
      data: {documentStatus: status},
    } = passport;

    const textProps = {
      italic: status === "PENDING",
      alert: status === "FAIL",
      theme: theme[globaltheme],
    };

    return (
      <Container>
        <StyledText {...textProps}>
          {hasRightToRent
            ? `${translations.rightToRent}${translations[status.toLowerCase()]}`
            : translations.uploadRight}
        </StyledText>
        {scannedByName && (
          <ScanningMetadata theme={theme[globaltheme]}>
            {this.getScanningMetadata()}
          </ScanningMetadata>
        )}
      </Container>
    );
  }
}

RightToRentPreview.propTypes = {
  passport: shape({
    data: shape({
      id: string,
      provider: oneOf(["TRUST_ID"]).isRequired,
      tenantId: string,
      documentStatus: oneOf(["PENDING", "SUBMITTED", "PASS", "FAIL"])
        .isRequired,
      seenBy: string.isRequired,
      seenAt: string.isRequired,
      status: oneOf(["GREEN", "YELLOW", "RED", "NONE"]).isRequired,
      documents: arrayOf(
        shape({
          id: string,
          type: oneOf(["PASSPORT", "BIOMETRIC_RESIDENCE_PERMIT"]),
          images: arrayOf(string).isRequired, // document refs
          expires: string,
        }),
      ).isRequired,
    }).isRequired,
    scannedByName: string,
    translations: shape({
      rightToRent: string.isRequired,
      uploadRight: string.isRequired,
      pending: string.isRequired,
      submitted: string.isRequired,
      pass: string.isRequired,
      fail: string.isRequired,
      seenBy: string.isRequired,
    }).isRequired,
  }).isRequired,
  theme: shape({
    textColor: string,
    errorColor: string,
  }).isRequired,
  getUser: func.isRequired,
  scannedByName: string,
  globaltheme: oneOf(["dark", "light"]),
};

const mapStateToProps = ({state: profile}) => ({
  scannedByName: profile && `${profile.firstName} ${profile.lastName}`,
});

const mapDispatchToProps = {
  getUser: fetchUserById,
};

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(themed(RightToRentPreview));
