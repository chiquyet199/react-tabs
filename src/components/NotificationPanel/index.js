import React from "react";
import {func, node, number, bool, string, oneOf} from "prop-types";
import {themed} from "../../utils/common";
import Icon from "../Icon";
import {Container, NotificationText, TextWrapper} from "./NotificationPanel";
import {
  NOTIFICATION_TYPES,
  NOTIFICATION_ICON_COLORS,
} from "../../constants/notifications";

export const NotificationPanel = ({
  text,
  content,
  textColor,
  iconColor,
  iconName,
  handlePress,
  type,
  theme,
  topmargin,
  bottommargin,
  isTopbar,
}) => {
  const colors = NOTIFICATION_ICON_COLORS(theme)[type];

  if (text || content) {
    return (
      <Container
        theme={theme}
        color={colors.bg}
        bottommargin={bottommargin}
        topmargin={topmargin}
        topbar={isTopbar && isTopbar.toString()}
      >
        {text && (
          <TextWrapper>
            <Icon
              name={iconName}
              size={26}
              color={iconColor || colors.iconColor}
            />
            <NotificationText
              theme={theme}
              color={textColor || iconColor || colors.text}
            >
              {text}
            </NotificationText>
            {handlePress && (
              <Icon
                name={iconName}
                size={26}
                color={colors.iconColor}
                onPress={handlePress}
                onClick={handlePress}
              />
            )}
          </TextWrapper>
        )}
        {content && content}
      </Container>
    );
  }

  return null;
};

NotificationPanel.defaultProps = {
  type: "",
};
NotificationPanel.propTypes = {
  handlePress: func,
  text: string,
  content: node,
  textColor: string,
  iconColor: string,
  iconName: string,
  type: oneOf(NOTIFICATION_TYPES),
  topmargin: number,
  bottommargin: number,
  isTopbar: bool,
};

export default themed(NotificationPanel);
