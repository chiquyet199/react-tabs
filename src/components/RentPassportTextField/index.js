import RentPassportTextField from "./RentPassportTextField";

export default RentPassportTextField;
