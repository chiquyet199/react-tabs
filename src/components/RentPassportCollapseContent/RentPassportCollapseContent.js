import React from "react";
import styled from "styled-components/native";
import {isEmpty} from "underscore";
import RentPassportTextField from "../RentPassportTextField";
import NotificationPanel from "../NotificationPanel";
import RentPassportHelpers from "../../utils/helpers/rentPassport";
import {colors} from "../../constants/theme";
import {SCORES, getScoreRanges} from "../../constants/credit-score";
import {
  arrayOf,
  array,
  func,
  object,
  oneOfType,
  string,
  oneOf,
} from "prop-types";
import RightToRentPreview from "../../components/RightToRentPreview";

export const FlexContainer = styled.View`
  flex-direction: row;
  width: 100%;
  flex-wrap: wrap;
`;

export const ScaleContainer = styled(FlexContainer)`
  justify-content: center;
  padding-vertical: 15px;
`;

export const FlexItemContainer = styled(FlexContainer)`
  border-bottom-color: ${({noborder}) =>
    noborder === "true" ? colors.none : colors.lightSteel};
  border-bottom-width: 1px;
  flex-grow: 2;
  width: 100%;
`;

export const RentPassportCollapseContent = ({
  status,
  messageGroups,
  multipleMessages,
  errorTexts,
  goTo,
  showModal,
  schema,
  globaltheme,
}) => {
  const isCreditStatusPoor = properties => {
    const [{creditScale}] = properties;
    const scoreRange = getScoreRanges(SCORES.VERY_POOR);
    const creditScaleLowerBound = scoreRange.from;
    const creditScaleUpperBound = scoreRange.to;

    return (
      creditScale &&
      creditScale.value > creditScaleLowerBound &&
      creditScale.value <= creditScaleUpperBound
    );
  };

  const parseMessages = (messagesToParse, creditWorthy) => {
    const failedMessage = "creditScore.failed";
    if (messagesToParse && messagesToParse.length) {
      const parsedMessages = messagesToParse.slice();
      if (!creditWorthy && parsedMessages.includes(failedMessage)) {
        parsedMessages.push(failedMessage);
      }

      return parsedMessages;
    }

    return null;
  };

  if (!isEmpty(schema)) {
    const actions = {goTo, showModal};
    const isCreditWorthy = !isCreditStatusPoor(schema);
    const componentStatus = isCreditWorthy ? status : "RED";
    const notificationMessages = parseMessages(messageGroups, isCreditWorthy);

    return (
      <FlexContainer>
        {notificationMessages && notificationMessages.length
          ? messageGroups.map(messageGroup => {
              const messagesInGroup = Array.isArray(messageGroup)
                ? messageGroup
                : [messageGroup];

              return messagesInGroup.map(message => (
                <NotificationPanel
                  key={message}
                  topmargin={5}
                  bottommargin={5}
                  content={RentPassportHelpers.renderContent(
                    componentStatus,
                    message,
                    errorTexts,
                    actions,
                  )}
                  type={
                    RentPassportHelpers.getNotificationType(componentStatus)
                      .type
                  }
                />
              ));
            })
          : null}
        {multipleMessages && multipleMessages.length
          ? multipleMessages.map(message =>
              message.errors.map(error => (
                <NotificationPanel
                  key={error}
                  topmargin={5}
                  bottommargin={5}
                  content={RentPassportHelpers.renderContent(
                    message.status,
                    error,
                    errorTexts,
                    actions,
                  )}
                  type={
                    RentPassportHelpers.getNotificationType(message.status).type
                  }
                />
              )),
            )
          : null}
        {Object.entries(schema).map(([key, property]) => {
          return (
            <FlexItemContainer
              key={key}
              noborder={String(Number(key) === schema.length - 1)}
            >
              {Object.entries(property).map(([objKey, obj]) => {
                if (obj.uiComponent === "rightToRentComponent") {
                  return (
                    <RightToRentPreview
                      key={key}
                      passport={obj}
                      globaltheme={globaltheme}
                    />
                  );
                }

                return (
                  <RentPassportTextField
                    key={`${objKey}`}
                    title={obj.title}
                    value={obj.value || ""}
                    width={obj.width}
                    fontSize={obj.fontSize}
                    fontStyle={obj.fontStyle}
                    align={obj.align}
                    error={obj.error}
                  />
                );
              })}
            </FlexItemContainer>
          );
        })}
      </FlexContainer>
    );
  }

  return null;
};

RentPassportCollapseContent.propTypes = {
  schema: oneOfType([array, object]).isRequired,
  status: string,
  messageGroups: arrayOf(oneOfType([string, object])),
  multipleMessages: arrayOf(oneOfType([string, object])),
  errorTexts: object,
  goTo: func,
  globaltheme: oneOf(["dark", "light"]).isRequired,
  showModal: func,
};

export default RentPassportCollapseContent;
