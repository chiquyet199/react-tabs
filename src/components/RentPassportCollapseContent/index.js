import {connect} from "react-redux";
import RentPassportCollapseContent from "./RentPassportCollapseContent";
import {go} from "../../actions/navigation";
import {setModalVisible} from "../../actions/modal";

const mapStateToProps = state => ({
  globaltheme: state.globaltheme,
});

const mapDispatchToProps = {
  goTo: go,
  showModal: setModalVisible,
};

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(RentPassportCollapseContent);
