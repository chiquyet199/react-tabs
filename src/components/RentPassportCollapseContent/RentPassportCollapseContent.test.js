import React from "react";
import {Platform} from "react-native";
import {shallow} from "enzyme";
import {initialState} from "../../__mocks__/initialState";
import translations from "../../utils/locales/en-US";
import RentPassportCollapseContent, {
  FlexContainer,
} from "./RentPassportCollapseContent";
import RentPassportTextField from "../RentPassportTextField";
import RightToRentPreview from "../../components/RightToRentPreview";
import NotificationPanel from "../NotificationPanel";
import {RentPassportCompleteContentSchema} from "../../schemas/subschemas/RentPassportCompleteContentSchema";
import RentPassportHelpers from "../../utils/helpers/rentPassport";
import configureStore from "redux-mock-store";

const componentName = "financial";
const compTranslations = translations.rent_passport_complete;

const schema = RentPassportCompleteContentSchema(
  componentName,
  initialState.rentPassport,
  compTranslations,
);

const aboutYou = {
  properties: () => [
    {
      name: {
        title: "test",
        value: "n/a",
        width: 100,
      },
      email: {
        title: "test",
        value: "",
        width: 100,
      },
      phoneNumber: {
        title: "test",
        value: "",
        width: 100,
      },
      rightToRentInfo: {
        documentStatus: "verified",
        legalWarnings: false,
        documentSample:
          "https://www.apostille.org.uk/sites/default/files/sample-passport700w-wm.gif",
        sanctions: false,
        status: "GREEN",
        uiComponent: "rightToRentComponent",
        messages: [],
      },
    },
  ],
};

const {status, messages} = initialState.rentPassport[componentName];

const mockProps = {
  schema: schema[componentName].properties(),
  globaltheme: "light",
  errorTexts: compTranslations.errors,
  status: "RED",
  messageGroups: ["grossAnnualIncome.unverified"],
  multipleMessages: undefined,
  scaleImages: {},
  goTo: () => {},
  showModal: () => {},
};

const mockPropsSchemaIsNull = {
  schema: {},
  globaltheme: "light",
  errorTexts: compTranslations.errors,
  status: "RED",
  messageGroups: [],
  multipleMessages: undefined,
  scaleImages: {},
  goTo: () => {},
  showModal: () => {},
};

const mockPropsWithoutNotificationPanel = {
  schema: schema[componentName].properties(),
  globaltheme: "light",
  errorTexts: compTranslations.errors,
  status: "GREEN",
  messageGroups: [],
  multipleMessages: undefined,
  scaleImages: {},
  goTo: () => {},
  showModal: () => {},
};

const mockPropsWithCustomSchema = {
  schema: aboutYou.properties(),
  globaltheme: "light",
  errorTexts: compTranslations.errors,
  status: "RED",
  messageGroups: [],
  multipleMessages: undefined,
  scaleImages: {},
  goTo: () => {},
  showModal: () => {},
};

const mockStore = configureStore(mockProps);
const Comp = <RentPassportCollapseContent {...mockProps} />;

const mockStoreNullSchema = configureStore(mockPropsSchemaIsNull);
const CompNullSchema = (
  <RentPassportCollapseContent {...mockPropsSchemaIsNull} />
);

const mockStoreCustomSchema = configureStore(mockPropsWithCustomSchema);
const CompCustomSchema = (
  <RentPassportCollapseContent {...mockPropsWithCustomSchema} />
);

const mockStoreWithoutNotification = configureStore(
  mockPropsWithoutNotificationPanel,
);
const CompWithoutNotification = (
  <RentPassportCollapseContent {...mockPropsWithoutNotificationPanel} />
);

describe("<RentPassportComplete />", () => {
  let wrapper;
  let hideNotificationPanel;
  let compSchemaNull;
  let compCustomSchema;
  beforeEach(() => {
    wrapper = shallow(Comp, {
      context: {store: mockStore(initialState)},
    });
    hideNotificationPanel = shallow(CompWithoutNotification, {
      context: {store: mockStoreWithoutNotification(initialState)},
    });
    compSchemaNull = shallow(CompNullSchema, {
      context: {store: mockStoreNullSchema(initialState)},
    });
    compCustomSchema = shallow(CompCustomSchema, {
      context: {store: mockStoreCustomSchema(initialState)},
    });
  });

  describe("RentPassportCollapseContent component", () => {
    it("renders properly - " + Platform.OS, () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("rightToRentPreviewComponent", () => {
    it("renders properly - " + Platform.OS, () => {
      expect(compCustomSchema.find(RightToRentPreview)).toExist();
    });
  });

  describe("Notification", () => {
    it("Notification panel must be null - " + Platform.OS, () => {
      const notification = hideNotificationPanel.find(NotificationPanel);
      expect(notification).toHaveLength(0);
    });

    it("renders properly - " + Platform.OS, () => {
      const actions = {goTo: mockProps.goTo, showModal: mockProps.showModal};
      const content = RentPassportHelpers.renderContent(
        mockProps.status,
        mockProps.messageGroups[0],
        mockProps.errorTexts,
        actions,
      );
      const {type} = RentPassportHelpers.getNotificationType(mockProps.status);
      const notification = wrapper.find(NotificationPanel);
      expect(notification).toHaveLength(1);
      expect(notification.prop("type")).toEqual(type);
      expect(JSON.stringify(notification.prop("content"))).toEqual(
        JSON.stringify(content),
      );
    });
  });

  it("displays Notification panel if notifications set", () => {
    const component = shallow(
      <RentPassportCollapseContent
        globaltheme="light"
        errorTexts={compTranslations.errors}
        schema={schema[componentName].properties()}
        messageGroups={["ERROR"]}
        status={status}
        goTo={() => {}}
        showModal={() => {}}
      />,
    );
    expect(component.find(NotificationPanel)).toHaveLength(1);
  });

  it("doesn't display Notification panel if notifications aren't set", () => {
    const component = shallow(
      <RentPassportCollapseContent
        globaltheme="light"
        errorTexts={compTranslations.errors}
        schema={schema[componentName].properties()}
        messageGroups={messages}
        status={status}
        goTo={() => {}}
        showModal={() => {}}
      />,
    );
    expect(component.find(NotificationPanel)).toHaveLength(0);
  });

  describe("Fields", () => {
    it(
      "Check RentPassportTextField match with schema title - " + Platform.OS,
      () => {
        const schemaItemProperties = mockProps.schema;
        const fields = wrapper.find(RentPassportTextField);

        fields.forEach(field => {
          const [fieldTitles] = Object.entries(schemaItemProperties).map(
            ([, property]) => {
              return Object.entries(property).map(
                ([, obj]) =>
                  obj.title === field.prop("title") ? obj.title : false,
              );
            },
          );
          const title = fieldTitles.find(i => i === field.prop("title"));
          if (!title) {
            expect(fieldTitles).toBe(9999);
            expect(field.props()).toBeFalsy();
          }
          expect(title).not.toBeUndefined();
        });
      },
    );

    it(
      "Check RentPassportTextField match with schema value - " + Platform.OS,
      () => {
        const schemaItemProperties = mockProps.schema;
        const fields = wrapper.find(RentPassportTextField);

        fields.forEach(field => {
          const [fieldTitles] = Object.entries(schemaItemProperties).map(
            ([, property]) => {
              return Object.entries(property).map(
                ([, obj]) =>
                  obj.value === field.prop("value") ? obj.value : false,
              );
            },
          );
          const title = fieldTitles.find(i => i === field.prop("value"));
          expect(title).not.toBeUndefined();
        });
      },
    );
  });

  describe("When schema properties is null", () => {
    it("renders properly - " + Platform.OS, () => {
      expect(compSchemaNull.find(FlexContainer)).not.toExist();
    });
  });
});
