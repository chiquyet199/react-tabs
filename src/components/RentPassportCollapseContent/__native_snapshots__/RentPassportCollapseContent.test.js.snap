// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RentPassportComplete /> RentPassportCollapseContent component renders properly - ios 1`] = `
<Styled(Component)>
  <Connect(WithTheme(NotificationPanel))
    bottommargin={5}
    content={
      <React.Fragment>
        <Icon
          allowFontScaling={false}
          color="#f1bf51"
          name="close-circle"
          size={26}
        />
        <Styled(Component)>
          <Styled(Text)
            color="#f1bf51"
          >
            Annual income not verified
          </Styled(Text)>
          <Styled(Text)
            color="#5ab88e"
            onPress={[Function]}
          >
            Verify income
          </Styled(Text)>
        </Styled(Component)>
      </React.Fragment>
    }
    key="grossAnnualIncome.unverified"
    topmargin={5}
    type="rent_passport_error"
  />
  <Styled(Styled(Component))
    key="0"
    noborder="true"
  >
    <RentPassportTextField
      key="totalAnnualIncome"
      title="Total annual income"
      value=" 100,000.00"
      width={50}
    />
    <RentPassportTextField
      error={false}
      key="activeCCJs"
      title="Active CCJs"
      value="None"
      width={50}
    />
    <RentPassportTextField
      key="previousCCJs"
      title="Previous CCJs"
      value="None"
      width={50}
    />
    <RentPassportTextField
      error={false}
      key="activeInsolvencies"
      title="Active Insolvencies"
      value="None"
      width={50}
    />
    <RentPassportTextField
      key="previousInsolvencies"
      title="Previous Insolvencies"
      value="None"
      width={50}
    />
    <RentPassportTextField
      error={false}
      key="bankrupt"
      title="Bankrupt"
      value="No"
      width={50}
    />
  </Styled(Styled(Component))>
</Styled(Component)>
`;
